Source/ARTMessage.m

```diff
@@ -45,3 +49,73 @@ - (NSInteger)messageSize {
 }
 
 @end
+
+@implementation ARTMessage (Decoding)
+
++ (instancetype)fromEncodedJsonObject:(NSDictionary *)json channelOptions:(ARTChannelOptions *)options error:(NSError **)error {
```

Comment:
I guess it should be `nullable` then.

Reply:
Maybe. I think we should add something to the spec.

Reply:
Searching for `.connectionId =` string across the Xcode project I didn't find any traces where described behavior could be implemented. `ARTJsonLikeEncoder/messageFromDictionary:` populates it:

`message.connectionId = [input artString:@"connectionId"]; `

Handling `@"connectionId"` being an empty string should be in another PR IMHO.

Reply:
Can't remember the exact thoughts at that moment, probably handling incoming JSON with absent fields is a more broad topic, but you are right, why not to take care about this particular field now. So, there are two options - make this field `nullable`, as the other fields in the `ARTBaseMessage` or return a decoding error. Since messages that are stored externally I guess have no any live connection, then the first option is I think the right one.

Reply:
I would agree, the former makes sense. Could we open a spec issue to get a discussion going and get something added to the spec?

Reply:
@lawrence-forooghian https://github.com/ably/docs/issues/1329